analyzer/runtime/evm/client.go

```diff
@@ -313,21 +317,40 @@ func EVMMaybeUnmarshalEncryptedData(data []byte, result *[]byte) (*EVMEncryptedD
 		if err := cbor.Unmarshal(*result, &callResult); err != nil {
 			return nil, fmt.Errorf("unmarshal outer call result: %w", err)
 		}
-		if callResult.IsUnknown() {
-			switch call.Format {
-			case sdkTypes.CallFormatEncryptedX25519DeoxysII:
-				var resultEnvelope sdkTypes.ResultEnvelopeX25519DeoxysII
+		switch call.Format {
+		case sdkTypes.CallFormatEncryptedX25519DeoxysII:
+			var resultEnvelope sdkTypes.ResultEnvelopeX25519DeoxysII
+			switch {
+			// The result is encrypted, in which case it is stored in callResult.Unknown
+			case callResult.IsUnknown():
 				if err := cbor.Unmarshal(callResult.Unknown, &resultEnvelope); err != nil {
 					return nil, fmt.Errorf("outer call result unmarshal unknown: %w", err)
 				}
-				encryptedData.ResultNonce = resultEnvelope.Nonce[:]
-				encryptedData.ResultData = resultEnvelope.Data
+			// The result may be unencrypted, in which case it is stored in callResult.Ok
+			//
+			// Note: IsUnknown() is already checked above, but we keep it here to make the
+			// logic explicit in the event of future refactors. IsSuccess() returns true both
+			// when the call succeeded and when it is unknown. However, we want this case
+			// to run iff the call succeeded.
+			case callResult.IsSuccess() && !callResult.IsUnknown():
+				if err := cbor.Unmarshal(callResult.Ok, &resultEnvelope); err != nil {
+					return nil, fmt.Errorf("outer call result unmarshal ok: %w", err)
+				}
+			// For non-evm txs as well as older Sapphire txs, the outer callResult may
+			// be Unknown and the inner callResult Failed. In this case, we extract the
+			// failed callResult.
+			case callResult.Failed != nil:
+				encryptedData.FailedCallResult = callResult.Failed
```

Comment:
per pro-wh's comment:
> it could be unknown outer + fail inner for non-evm transactions and for old transactions

Adjusted to extract the `FailedCallResult` and bubble it up to the calling fn in `extract.go`, which in turn uses the failed call result to populate the error fields.